public/src/main/java/com/revenuecat/purchases/Package.kt

```diff
@@ -9,14 +9,21 @@ import kotlinx.parcelize.Parcelize
  * @property identifier Unique identifier for this package. Can be one a predefined package type or a custom one.
  * @property packageType Package type for the product. Will be one of [PackageType].
  * @property product [StoreProduct] assigned to this package.
- * @property offering offeringID this package was returned from.
+ * @property offering offering this package was returned from.
+ * @property subscriptionPeriod Google BC5 only. String representation of the subscription period, used for filtering ProductDetails
+ * for the proper base plan to create a StoreProduct. For now, we are not worrying if it matches PackageType
+ * @property storeProductIdentifier  BC4/Amazon sku, BC4 subscription ID. used for querying the store.
  */
 @Parcelize
 data class Package(
     val identifier: String,
     val packageType: PackageType,
     val product: StoreProduct,
-    val offering: String
+    val offering: Offering,
+
+    // TODO both of these are only used to create StoreProduct, maybe we don't need to store them in this class
+    private val subscriptionPeriod: String?,
```

Comment:
not sure about this - there's standard stuff for parsing a iso8601 duration, like https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html#parse-java.lang.CharSequence- but i'm not sure whether something like this gives a benefit. Most common use case would be to print out some human readable text, but that doesn't seem to be provided.

Reply:
+1 for some form of data class or something to encapsulate. 
Flexibility in how you show it to humans is important - for example, you could have this: 
- price: 39.99
- period: 1 year
And show: 
"**$3.33/month** _($39.99/year_)"

This is a fairly common use case and it was a big complaint for StoreKit 2, which didn't provide a way to do this

Reply:
@swehner maybe we can make a ticket to think about this? i feel like the scope of this PR is getting pretty large.

Reply:
I agree, with @beylmk

Reply:
@aboedo coming back to this comment WAYYY later... do you have any thoughts on how to localize this kind of helper function?

Reply:
uhhhmm... well if we _need_ to provide a string, we need to format it with the Store's locale if possible. 

Since you could have your phone set to locale: US, but the store account: Uruguay (that's how I have it for example). 

Do we need to do it, though? I have very little context here, but I don't think BillingClient exposes the locale for the user's account. If that's the case, then there's no way to get an accurate string other than bridging the one that BillingClient exposes directly

Reply:
BC does expose currency IIRC, but even that isn't entirely guaranteed to help - you could have `€ 1,00` and `1.00 €` represent the same value in different locales with the same currency

Reply:
BC exposes a `formattedPrice` which supposedly handles the ccurrency ases you mentioned... it's more the "per year" text that i am unsure about. 

we don't need to do it, i thought it was just something people asked about in StoreKit2 that would be a nice addition.  so i just wanted to check whether you had thought about potential implementations since then... I'm just going through old TODOs to make sure we at least have tickets for potential improvements.

Reply:
sounds good. I'd say we don't need it for v1, we can circle back to it if customers start asking about it